### PR TITLE
Manifest Command Unit Tests

### PR DESCRIPTION
**Pull request by astarinmymind**
One line in logManifestError is not covered because it is never called by `manifest.ts` . This is because the error parameter is optional and it is never used throughout the program.

### CODE REVIEW

File: src/cmds/manifest/manifest.ts [hunk 1]
```diff
@@ -129,48 +129,50 @@ export async function manifest(argv: YargsArgs): Promise<void> {
   }
 
   // check web3Wallet properties
-  const { bundle, initialPermissions } = pkg.web3Wallet || {};
-  if (bundle?.local) {
-    if (!(await isFile(bundle.local))) {
-      logManifestError(`'bundle.local' does not resolve to a file.`);
+  if (pkg.web3Wallet !== undefined) {
```
Comment (rekmarks) [inline src/cmds/manifest/manifest.ts:132]: In a follow-up PR, we should probably create an `else` case for this and log a warning or possibly an error.

Comment (rekmarks) [reply src/cmds/manifest/manifest.ts:132]: Btw, what motivated this change?

File: test/cmds/manifest/manifest.test.js [hunk 2]
```diff
@@ -0,0 +1,318 @@
+/* eslint-disable jest/prefer-strict-equal */
```
Comment (rekmarks) [inline test/cmds/manifest/manifest.test.js:1]: As noted in other reviews, we want to use strict equal whenever feasible.

File: test/cmds/manifest/manifest.test.js [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+  return {
+    bundle: {
+      local: 'dist/bundle.js',
+      url: 'http://localhost:8081/dist/bundle.js',
+    },
+    initialPermissions: { alert: {} },
+  };
+};
+
+const getPackageJson = async (
```
Comment (rekmarks) [inline test/cmds/manifest/manifest.test.js:25]: I don't believe that this needs to be `async`.

File: test/cmds/manifest/manifest.test.js [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+        .mockImplementationOnce(async () => await getPackageJson());
+      await expect(manifest(mockArgv)).rejects.toThrow('Invalid parsed package.json: false');
+      expect(fsMock).toHaveBeenCalledTimes(1);
+    });
+
+    it('checks presence of required and recommended keys', async () => {
+      const badJSON = {};
+      jest.spyOn(JSON, 'parse').mockImplementation((value) => value);
+      jest.spyOn(console, 'warn').mockImplementation();
+      jest.spyOn(console, 'log').mockImplementation();
```
Comment (rekmarks) [inline test/cmds/manifest/manifest.test.js:95]: These are repeated in almost every test case, and seem like a great candidate for a `beforeEach` hook.

File: test/cmds/manifest/manifest.test.js [hunk 5]
```diff
@@ -0,0 +1,318 @@
...
+        global.snaps.suppressWarnings = true;
+        const badJSON = {};
+        jest.spyOn(JSON, 'parse').mockImplementation((value) => value);
+        jest.spyOn(console, 'warn').mockImplementation();
+        jest.spyOn(console, 'log').mockImplementation();
+        jest.spyOn(fs.promises, 'readFile')
+          .mockImplementationOnce(async () => badJSON);
+        await manifest(mockArgv);
+        expect(global.console.warn).not.toHaveBeenCalled();
+      });
```
Comment (rekmarks) [inline test/cmds/manifest/manifest.test.js:247]: The name of this test is somewhat unclear to me. Is it checking whether manifest warnings are suppressed? If so, a better name may be: `suppresses logManifestWarning per global settings`

**Review by rekmarks — CHANGES_REQUESTED** [review]
Now you're cooking with gas! 😄 

One general comment in addition to the inline: would it be possible to spy on e.g. `logManifestWarning` and `logManifestError` instead `console.warn` and `console.error`? This is a genuine question; we don't want to do that at the expense of too much extra work.

File: src/cmds/manifest/manifest.ts (hunk 1 above)
Comment (astarinmymind) [reply src/cmds/manifest/manifest.ts:132]: There was a typescript complaint somewhere bubbled down inside.

**Comment by astarinmymind** [comment]
> Now you're cooking with gas! 😄
> 
> One general comment in addition to the inline: would it be possible to spy on e.g. `logManifestWarning` and `logManifestError` instead `console.warn` and `console.error`? This is a genuine question; we don't want to do that at the expense of too much extra work.

This is because `logManifestWarning` and `logManifestError` are internal functions. SpyOn is used to mock functions from other modules.

File: src/cmds/manifest/manifest.ts (hunk 1 above)
Comment (rekmarks) [reply src/cmds/manifest/manifest.ts:132]: Cool. Issue: #96

File: test/cmds/manifest/manifest.test.js (hunk 4 above)
Comment (astarinmymind) [reply test/cmds/manifest/manifest.test.js:95]: Only console error is repeated across 90%, so I will move the console error mock to a `beforeEach` hook.

**Comment by rekmarks** [comment]
> This is because `logManifestWarning` and `logManifestError` are internal functions. SpyOn is used to mock functions from other modules.

Oh, I thought they were in a different file. That's fine then, let's just keep things as they are!

**Review by rekmarks — APPROVED** [review]
LGTM!

**Pull request merged**